plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/ConfigSetup.java

```diff
@@ -0,0 +1,162 @@
+            cfg.setPluginTxt(Settings.DEFAULT_PLUGIN_TXT);
+        } else {
+            System.out.println("File containing list of plugins to be downloaded: " + options.getPluginTxt());
+            cfg.setPluginTxt(options.getPluginTxt());
+        }
+
+        if (Files.exists(cfg.getPluginTxt().toPath())) {
+            try {
+                Scanner scanner = new Scanner(cfg.getPluginTxt(), StandardCharsets.UTF_8.name());
+                System.out.println("Reading in plugins from " + cfg.getPluginTxt().toString() + "\n");
+                while (scanner.hasNextLine()) {
+                    Plugin plugin = parsePluginLine(scanner.nextLine());
+                    plugins.add(plugin);
+                }
+            } catch (FileNotFoundException e) {
+                System.out.println("Unable to open " + cfg.getPluginTxt());
+            }
+        } else {
+            System.out.println(cfg.getPluginTxt() + " file does not exist");
+        }
+
+        cfg.setPlugins(plugins);
+    }
+
+    public Plugin parsePluginLine(String pluginLine) {
+        String[] pluginInfo = pluginLine.split(":");
+        String pluginName = pluginInfo[0];
+        String pluginVersion = "latest";
+        String pluginUrl = null;
+        if (pluginInfo.length >= 2) {
+            pluginVersion = pluginInfo[1];
+        }
+        if (pluginInfo.length == 3) {
+            pluginUrl = pluginInfo[2];
+        }
+
+        return new Plugin(pluginName, pluginVersion, pluginUrl);
+    }
+
+    public void getPluginDir() {
+        if (options.getPluginDir() == null) {
+            System.out.println("No directory to download plugins entered. " +
+                    "Will use default of " + Settings.DEFAULT_PLUGIN_DIR);
+            cfg.setPluginDir(Settings.DEFAULT_PLUGIN_DIR);
+        } else {
+            System.out.println("Plugin download location: " + options.getPluginDir());
+            cfg.setPluginDir(options.getPluginDir());
+        }
+    }
+
+
+    public void getWar() {
+        if (options.getJenkinsWar() == null) {
+            System.out.println("No war entered. Will use default of " + Settings.DEFAULT_JENKINS_WAR);
+            cfg.setJenkinsWar(Settings.DEFAULT_JENKINS_WAR);
+        } else {
+            System.out.println("Will use war file: " + options.getJenkinsWar());
+            cfg.setJenkinsWar(options.getJenkinsWar());
+        }
+    }
+
+
+    public void getUpdateCenters() {
+        String jenkinsUc = "";
+        if (!StringUtils.isEmpty(options.getJenkinsUc())) {
+            jenkinsUc = options.getJenkinsUc();
+            System.out.println("Using update center " + jenkinsUc + " specified with CLI option");
+        } else if (!StringUtils.isEmpty(System.getenv("JENKINS_UC"))) {
+            jenkinsUc = System.getenv("JENKINS_UC");
+            System.out.println("Using update center " + jenkinsUc + " from JENKINS_UC environment variable");
+        } else {
+            jenkinsUc = "https://updates.jenkins.io";
```

Comment:
Try to avoid hard coded URLs if at all possible. They will change at some point in history and that would force the maintainer to release a new version of this plugin.

Reply:
This url is highly unlikely to change and is configurable via env variable and CLI, I would suggest extracting it out to a Constants / Configuration class but this is a default that is very reasonable imo...

Reply:
TBH: the default `updates.jenkins.io` should probably be a `public final` variable anyway.  And would it make more sense to store the defaults in the `Config` object?  That way if something's not defined and the user is coming in from a different path than the cli, the PluginManager won't throw errors.